food-truck-frontend/src/pages/truck/edit/[truckId].tsx

```diff
@@ -1,140 +1,166 @@
+    message: "",
+    tags: [],
+    menuContentType: null,
+  });
+
+  const onMenuChange = (event: React.FormEvent) => {
+    const formData = new FormData();
+    const imagefile = document.querySelector('#fileInput') as HTMLInputElement;
+
+    if (imagefile === null || imagefile.files === null || imagefile.files[0] == null) {
+      console.log("Don't see a file there mate");
+      return;
     }
 
-  render() {
-    if (this.state.userId == -1)
-      return <CircularProgress/>
-      
-    return (
-        <>
-            <Typography variant={'h4'}>Edit Truck ID: {this.state.id}</Typography>
-            <Form submitMethod="PUT" submitUrl={'/truck/update'} onSuccessfulSubmit={this.onSubmit}
-                  onFailedSubmit={this.onFail}>
-                <TextField className={"hidden"} disabled label="Truck ID" name="truckId" defaultValue={this.state.id}/>
-
-                <TextField label="Truck Name" name="name" defaultValue={this.state.name}/>
-                <TextField label="Description" name="description"
-                           defaultValue={this.state.description}/>
-                <TextField label="Price Rating" name="priceRating"
-                           defaultValue={this.state.priceRating}/>
-                <MultiField
-                  title="Truck Tags"
-                  name="tags"
-                  variant="h4"
-                  value={this.state.tags}
-                />
-            </Form>
-
-            <Typography variant={'h4'}>Upload Menu</Typography>
-            <form encType="multipart/form-data">
-                <Button component="label">Upload Menu
-                    <input className="hidden" type="file" id="fileInput" name="file" onChange={this.onMenuChange}/>
-                </Button>
-            </form>
-
-            <Button color="secondary"
-                    onClick={this.deleteTruck}>
-                Delete Truck
-            </Button>
-            <br/>
-            {this.state.message}
-        </>
-    );
+    formData.append("file", imagefile.files[0]);
+    api.post(`/truck/${state.id}/upload-menu`, formData, {
+      headers: {
+        'Content-Type': 'multipart/form-data'
+      }
+    })
+      .then(_ => setState({...state, message: "Successfully uploaded menu"}))
+      .catch(err => setState({...state, message: "Failed to upload menu"}));
   }
 
-    onMenuChange = (event: React.FormEvent) => {
-        var formData = new FormData();
-        var imagefile = document.querySelector('#fileInput') as HTMLInputElement;
-
-        if (imagefile === null || imagefile.files === null || imagefile.files[0] == null) {
-            console.log("Don't see a file there mate");
-            return;
-        }
-
-        formData.append("file", imagefile.files[0]);
-        api.post(`/truck/${this.state.id}/upload-menu`, formData, {
-            headers: {
-                'Content-Type': 'multipart/form-data'
-            }
+  useEffect(() => {
+    if (state.userId === -1) {
```

Comment:
I'm confused by this check, given that it only sees if it cab edit the truck if the ID is an invalid placeholder ID. Or am I misreading this code?

Reply:
Yeah this seemed a bit error prone when I looked at it as well. The empty truck function sets the userId to 0 anyways, so I changed this `useEffect` to depend on the truckId in the props so it only runs when that changes, which lets us get rid of the if statement..